arrows/core/metadata_map_io_csv.cxx

```diff
@@ -24,6 +31,49 @@ namespace arrows {
 
 namespace core {
 
+namespace {
+
+// ----------------------------------------------------------------------------
+template < typename out >
+void
+split( std::string const& s, char delim, out result )
+{
+  std::istringstream iss( s );
+  std::string item;
+
+  while( std::getline( iss, item, delim ) )
```

Comment:
I have reservations whether this is really the most efficient implementation. I suspect something based on `substr` would be better. (Too, such an implementation might eventually be made to use `string_view`, whereas I'm not sure that's possible with `istringstream`.)